cluster/addons/fluentd-gcp/fluentd-gcp-configmap-old.yaml

```diff
@@ -0,0 +1,425 @@
+    # entries that are up to 100KB in size.
+    <filter kubernetes.**>
+      @type record_transformer
+      enable_ruby true
+      <record>
+        log ${record['log'].length > 100000 ? "[Trimmed]#{record['log'][0..100000]}..." : record['log']}
+      </record>
+    </filter>
+
+    # Do not collect fluentd's own logs to avoid infinite loops.
+    <match fluent.**>
+      @type null
+    </match>
+
+    # We use 2 output stanzas - one to handle the container logs and one to handle
+    # the node daemon logs, the latter of which explicitly sends its logs to the
+    # compute.googleapis.com service rather than container.googleapis.com to keep
+    # them separate since most users don't care about the node logs.
+    <match kubernetes.**>
+      @type google_cloud
+
+      # Try to detect JSON formatted log entries.
+      detect_json true
+      # Collect metrics in Prometheus registry about plugin activity.
+      enable_monitoring true
+      monitoring_type prometheus
+      # Allow log entries from multiple containers to be sent in the same request.
+      split_logs_by_tag false
+      # Set the buffer type to file to improve the reliability and reduce the memory consumption
+      buffer_type file
+      buffer_path /var/log/fluentd-buffers/kubernetes.containers.buffer
+      # Set queue_full action to block because we want to pause gracefully
+      # in case of the off-the-limits load instead of throwing an exception
+      buffer_queue_full_action block
+      # Set the chunk limit conservatively to avoid exceeding the recommended
+      # chunk size of 5MB per write request.
+      buffer_chunk_limit 1M
+      # Cap the combined memory usage of this buffer and the one below to
+      # 1MiB/chunk * (6 + 2) chunks = 8 MiB
+      buffer_queue_limit 6
+      # Never wait more than 5 seconds before flushing logs in the non-error case.
+      flush_interval 5s
+      # Never wait longer than 30 seconds between retries.
+      max_retry_wait 30
+      # Disable the limit on the number of retries (retry forever).
+      disable_retry_limit
+      # Use multiple threads for processing.
+      num_threads 2
+      use_grpc true
+    </match>
+
+    # Keep a smaller buffer here since these logs are less important than the user's
+    # container logs.
+    <match **>
+      @type google_cloud
+
+      detect_json true
+      enable_monitoring true
+      monitoring_type prometheus
+      # Allow entries from multiple system logs to be sent in the same request.
+      split_logs_by_tag false
+      detect_subservice false
+      buffer_type file
+      buffer_path /var/log/fluentd-buffers/kubernetes.system.buffer
+      buffer_queue_full_action block
+      buffer_chunk_limit 1M
+      buffer_queue_limit 2
+      flush_interval 5s
+      max_retry_wait 30
+      disable_retry_limit
+      num_threads 2
+      use_grpc true
+    </match>
+metadata:
+  name: fluentd-gcp-config-v1.2.5
```

Comment:
Can these configmaps be named differently? If the name stays the same, it will be harder to inspect the objects - we should be able to tell which one is used, just by looking at the name. Trying to infer this from configmap contents will be tedious. How about fluentd-gcp-legacy-config? or ...-old-config?

Reply:
Good point. Changed.